anonymization_PII_numbers.py

```diff
@@ -0,0 +1,130 @@
+
+    re_ordinal = re.compile(r"([0-9]{1,2})(nd|th|rd)")
+    df["utterance_with_ordinal"] = df["utterances"].apply(lambda x: True if re.findall(re_ordinal,x) else False)
+
+    re_4_digit = re.compile(r"\b(?:\d[- ]?){3}\d\b")
+    df["utterance_with_4_digit_between_1970_and_2030"] = df["utterances"].apply(lambda x: True if set(re.findall(re_4_digit,x)).intersection(set([str(i) for i in range(1970,2031)])) else False)
+
+    df["utterance_with_4_digit"] = df["utterances"].apply(lambda x: True if re.findall(re_4_digit,x) else False)
+
+    re_threedigit = re.compile(r"[^0-9]?(?:\d[- ]?){2}\d[^0-9]?")
+    df["utterance_with_three_digit_and_cvv"] = df["utterances"].apply(lambda x: True if ((x.lower().find("cvv") != -1) and re.findall(re_threedigit,x)) else False)
+
+    df["utterance_with_three_digit"] = df["utterances"].apply(lambda x: True if re.findall(re_threedigit,x) else False)
+
+    re_twodigit = re.compile(r"[^0-9]?([0-9]{2})[^0-9]?")
+    df["utterance_with_two_digits"] = df["utterances"].apply(lambda x: True if re.findall(re_twodigit,x) else False)
+    
+    # re_singledigit = re.compile(r"\b[0-9]\b")
+    re_singledigit = re.compile(r"[^0-9]?([0-9])[^0-9]?")
+    df["utterance_with_single_digits"] = df["utterances"].apply(lambda x: True if re.findall(re_singledigit,x) else False)
+    
+    classes = df.columns.tolist()
+    classes.remove("utterances")
+    classes.remove("utterance_with_digits")
+    classes.remove("anonymized_phrases")
+
+    df["class"] = df.apply(assign_class,args=[classes],axis=1)
+    print(df[["utterances","class"]].groupby(["class"]).count())
+
+    df["anonymized_phrases"] = df["utterances"]
+
+    df.loc[df["class"]=="utterance_more_than_16_digit","anonymized_phrases"] = df.loc[df["class"]=="utterance_more_than_16_digit","utterances"].apply(replacement,args=[re_more_than_16_digit])
+
+    df.loc[df["class"]=="utterance_16_digit","anonymized_phrases"] = df.loc[df["class"]=="utterance_16_digit","utterances"].apply(replacement,args=[re_16_digit])
+
+    df.loc[df["class"]=="utterance_with_more_than_4_digit","anonymized_phrases"] = df.loc[df["class"]=="utterance_with_more_than_4_digit","utterances"].apply(replacement,args=[re_more_than_4_digit])
+
+    output_csv = input.replace(".txt",".csv")
+    df.to_csv(output_csv,sep=",",encoding="utf",index=False)
+    print(f"\nSize of dataframe after processing: {df.shape}")
+    print(f"Processed dataset is stored as CSV in {output_csv}")
+
+    output_anonymized = input.replace("_cleansed.txt","_anonymized.txt")
+    anonymized_phrases = df["anonymized_phrases"].to_list()
+    anonymized_phrases = "\n".join(anonymized_phrases)
+    with open(output_anonymized, mode="w", encoding="utf8") as f:
+        f.write(anonymized_phrases)
+    print(f"Anonymized phrases are stored in {output_anonymized}")
+    
+
+def replacement(utterance: str,pattern: re) -> str:
+    """performs random shuffling of numbers"""
+
+    re_digits = re.compile(r"\b[0-9]+\b")
```

Comment:
I think this regex makes assumptions about the input regex which aren't valid - i.e that it will obey word boundaries.  I think it should just be any digit.